### ISSUE TITLE
Proton 5.13-1 broken monitor handling

### ISSUE THREAD

**Issue by BieHDC** [ValveSoftware/Proton#4275]
Affected games so far: Battlefield Bad Company 2, GTA 4, GTA SA, GTA V (testing the library in alphabetical order)

I have a dual monitor setup where i have a center screen and a left of center screen. In Proton 5.0-9 it would spawn the game on my center screen which is also set as primary, but in Proton 5.13-1 it now spawns on the left monitor, however the mouse region lock would be applied to the center screen. 
On the example of GTA 4 and GTA V, it would spawn the launcher window on the center screen, but when the game starts, it fullscreens on the left screen while the mouse is locked to the center screen. Doing alt+tab twice would fix the mouse region to the left screen.
The other issue is that when alt+enter'ing to windowed mode, with the goal of manually fixing the monitor used, it will instead just minimise the game and when reopening it, it switches back to fullscreen immediately.

What do you require from me to debug this issue?

**Comment by BieHDC** [comment]
Hitman 2 lets me go to windowed mode and move the game window, but it will maximise on the left monitor again
Just Cause 3 is the same as Hitman 2, but the game thinks its started minimised (double alt+tab fixes that)
Tomb Raider 2013 (forcing the use of proton) which lets you choose the screen to use has no effect on where the window will be

**Comment by alasky17** [comment]
@BieHDC Could you give some details on your setup?  Attaching the output of System Information is a good start.  Also any other WM details that are not included in that output could be relevant.  And what is the resolution on both of your monitors?

**Comment by BieHDC** [comment]
> @BieHDC Could you give some details on your setup? Attaching the output of System Information is a good start. Also any other WM details that are not included in that output could be relevant. And what is the resolution on both of your monitors?

I am using Artix Linux with the MATE desktop, no compositor or anything else fancy, default setup. Both Monitors are 1920x1080.

System Info
https://pastebin.com/H0huwgQa

**Comment by zzhiyi** [comment]
Looking into this.

**Comment by BieHDC** [comment]
> Looking into this.

Still an issue in 5.13-3

**Comment by alasky17** [comment]
@BieHDC Could you try deleting the prefix for one of the affected games (preferably GTA V)?  If you have cloud saves enabled, you shouldn't lose any progress, but you can just move the prefix to 271590.bak instead to make sure you don't lose any data.  The standard location is ~/.steam/steam/steamapps/compatdata/271590

I installed MATE and I was seeing the exact behavior that you described with 5.13-6 with GTA V.  I then deleted the prefix as part of the testing, and it fixed the issue and now I cannot replicate it, and the game is launching to the correct monitor.

**Comment by BieHDC** [comment]
> @BieHDC Could you try deleting the prefix for one of the affected games (preferably GTA V)? If you have cloud saves enabled, you shouldn't lose any progress, but you can just move the prefix to 271590.bak instead to make sure you don't lose any data. The standard location is ~/.steam/steam/steamapps/compatdata/271590

Didnt fix the issue, at least for gta 5. The long story is that for fullscreen, the ingame which monitor to use setting still has no effect, but it remembers the correct monitor for windowed mode. And alt+enter switching between fullscreen and windowed mode now works. And going to windowed mode, moving the game window and trying to switch back to fullscreen on the correct monitor also does not work.

**Comment by alasky17** [comment]
@BieHDC I may have found the root of the difference.  On 5.13-6 - With GTAV, the game is on the correct monitor when I am in dx11 mode, but when I switch to dx10 mode, the game goes to the correct monitor.  On 5.0-10 - The game is only in dx10 mode (I can't get it to switch), and thus on the correct monitor.  Can you confirm that switching to dx10 on 5.13-6 fixes the game so its on the correct monitor?

**Comment by BieHDC** [comment]
> @BieHDC I may have found the root of the difference. On 5.13-6 - With GTAV, the game is on the correct monitor when I am in dx11 mode, but when I switch to dx10 mode, the game goes to the correct monitor. On 5.0-10 - The game is only in dx10 mode (I can't get it to switch), and thus on the correct monitor. Can you confirm that switching to dx10 on 5.13-6 fixes the game so its on the correct monitor?

No, cant confirm. Using DX10 or 10.1 does not fix the issue, it has the same behaviour as DX11.

**Comment by alasky17** [comment]
@BieHDC Thank you for checking.  Looks like that was a false lead :/  Could you also try launching the game with the minimum resolution (800x600 on my setup)?  I found that sometimes GTAV would automatically use that resolution, and when it did so, it would launch on the primary monitor.  I could not figure out when it decided to keep the previously set resolution vs setting a new resolution on launch, so it might be hard to launch with this resolution.

**Comment by zzhiyi** [comment]
Is it still an issue on proton 6.3-8. I tried Battlefield Bad Company 2 on it and it seems fine now.
